### PR TITLE
Unit Testing Added - unit.service.spec.ts and cart.service.spec.ts

### PR DESCRIPTION
**Pull request by shubhwale**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -11,17 +11,19 @@ import { EditBookComponent } from './books/edit-book/edit-book.component';
...
-  {path : 'user',component : UserComponent,
-  children :[
-    {path : "registration",component : RegistrationComponent}, //PATH /user/registration
-    {path : "login",component : LoginComponent}
-  ]},
-  {path : 'books',component : BooksListComponent},
-  {path : 'book/:id',component : BookDetailsComponent},
-  {path : 'add_book',component : AddBookComponent},
-  {path : 'edit_book/:id',component : EditBookComponent},
-  {path : 'cart',component : CartComponent,canActivate:[AuthGuard]}
```
Comment (ketannaik) [inline src/app/app-routing.module.ts:24]: Why is router file changed while writing the test cases?

File: src/app/shared/user.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import { TestBed } from '@angular/core/testing';
+import { UserService } from './user.service';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { Observable } from 'rxjs';
+
+fdescribe('UserService', () => {
```
Comment (ketannaik) [inline src/app/shared/user.service.spec.ts:6]: use describe instead of fdescribe. fdescribe will run only that test case nothing else.

**Comment by ketannaik** [comment]
Overall the PR looks good only the formatting changes should not be done along with it. PR is just for unit test case. Formatting could have been other PR.

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (shubhwale) [reply src/app/app-routing.module.ts:24]: Yeah this was an unintended formatting change by my side. My bad. Won't happen again

File: src/app/shared/user.service.spec.ts (hunk 2 above)
Comment (shubhwale) [reply src/app/shared/user.service.spec.ts:6]: done

**Pull request merged**